src/3p-frame.js

```diff
@@ -256,6 +258,22 @@ function getCustomBootstrapBaseUrl(parentWindow, opt_strictForUnitTest) {
 }
 
 /**
+ * Returns a randomized sentinel value for 3p iframes.
+ * The format is "%d-%d" with the first value being the depth of current
+ * window in the window hierarchy and the second a random integer.
+ * @param {!Window} parentWindow
+ * @return {string}
+ * @visibleForTesting
+ */
+export function generateSentinel(parentWindow) {
+  let windowDepth = 0;
+  for (let win = parentWindow; win != window.top; win = win.parent) {
+    windowDepth++;
+  }
+  return '' + windowDepth + '-' + Math.round(Math.random() * 1000000);
```

Comment:
You could just share this code:
https://github.com/ampproject/amphtml/blob/master/src/3p-frame.js#L216